src/backend/distributed/commands/table.c

```diff
@@ -385,6 +408,11 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand)
 	 */
 	ErrorIfAlterTableDefinesFKeyFromPostgresToCitusLocalTable(alterTableStatement);
 
+	if (AlterTableHasCommandByFunc(alterTableStatement, AlterTableCmdAddsOrDropsFkey))
```

Comment:
Do we need `AlterTableHasCommandByFunc ` and the function pointers? We are only using it  with `AlterTableCmdAddsOrDropsFkey `, and it sounds more straight forward to call it directly?

Reply:
yes, but we will use the same function with `AlterTableCmdAddsFkey` & `AlterTableCmdDropsFkey` when implementing auto conversion for add/drop fkey commands, so I wanted to keep it reusable